### PR TITLE
Feat/colors 2020 badge

### PR DESCRIPTION
**Pull request by EdwardIrby**

### CODE REVIEW

**Comment by danethurber** [comment]
Screen Shot 2020-01-27 at 3 17 07 PM
Screen Shot 2020-01-27 at 3 16 49 PM

@EdwardIrby @sweeting color contrast seems off

**Comment by EdwardIrby** [comment]
> Screen Shot 2020-01-27 at 3 17 07 PM
> Screen Shot 2020-01-27 at 3 16 49 PM
> 
> @EdwardIrby @sweeting color contrast seems off

The second one minus the gray is right You''re comparing subtle on light to the default badge on light. @sweeting need to know what to do about the dark badge on light I missed that one. It's background is `rgba(176, 176, 176, 0.25);` on Dark what about on light theme?

File: packages/badge/src/css/index.js [hunk 1]
```diff
@@ -11,82 +19,87 @@ export default {
...
-    color: core.colors.gray04,
-    backgroundColor: transparentize(0.85, core.colors.yellow)
+    color: colorsTextIcon.highOnLight,
+    backgroundColor: colorsYellow[1],
+    borderColor: colorsYellow[1]
   },
   [subtleThemeWithColor(themeNames.light, colors.red)]: {
-    color: core.colors.gray04,
-    backgroundColor: transparentize(0.85, core.colors.red)
+    color: colorsTextIcon.highOnDark,
```
Comment (jaketrent) [inline packages/badge/src/css/index.js]: After seeing more comments and becoming more careful, I do see something that looks odd to my eye, and checking storybook does not seem quite right:

That is, red subtle on Light theme.

Screen Shot 2020-01-27 at 4 06 06 PM

Comment (jaketrent) [reply packages/badge/src/css/index.js]: Think this should be highOnLight.

**Comment by sweeting** [comment]
On the gray badge, the background is rgba(176, 176, 176, 0.25) on both dark & light theme. The text color on light theme is textIconOnLightHigh, and on dark theme the text color is textIconOnDarkHigh.

The text color on the red subtle badge on light theme should be textIconOnDarkHigh.

File: packages/badge/src/css/index.js [hunk 2]
```diff
@@ -11,82 +19,87 @@ export default {
...
+    lineHeight: type.lineHeightStandard,
+    padding: `0 ${layout.spacingXSmall}`,
     textTransform: 'uppercase'
   },
 
   [defaultWithColor(colors.gray)]: {
-    color: core.colors.white,
-    backgroundColor: core.colors.gray03,
-    borderColor: core.colors.gray03
+    color: colorsTextIcon.highOnDark,
```
Comment (jaketrent) [inline packages/badge/src/css/index.js]: I think there's a no variation not taken into account here.

Default appearance, color gray, light theme.  

highOnLight should be the text color there.

**Pull request merged**